include/openthread/link.h

```diff
@@ -306,226 +306,291 @@ OTAPI void OTCALL otLinkSetPollPeriod(otInstance *aInstance, uint32_t aPollPerio
+ * This method adds an Extended Address to the filter.
  *
- * @param[in]  aInstance A pointer to an OpenThread instance.
- * @param[in]  aExtAddr  A pointer to the IEEE 802.15.4 Extended Address.
+ * @param[in]  aInstance    A pointer to an OpenThread instance.
+ * @param[in]  aExtAddress  A reference to the Extended Address.
+ *
+ * @retval OT_ERROR_NONE           Successfully added @p aExtAddress to the Filter.
+ * @retval OT_ERROR_ALREADY        If @p aExtAddress was already in the Filter.
+ * @retval OT_ERROR_INVALID_ARGS   If @p aExtAddress is NULL.
+ * @retval OT_ERROR_NO_BUFS        No available entry exists.
+ *
+ * @sa otLinkFilterGetAddressMode
+ * @sa otLinkFilterSetAddressMode
+ * @sa otLinkFilterAddAddressRssIn
+ * @sa otLinkFilterRemoveAddress
+ * @sa otLinkFilterClearAddresses
+ * @sa otLinkFilterGetNextAddress
+ * @sa otLinkFilterAddRssIn
+ * @sa otLinkFilterRemoveRssIn
+ * @sa otLinkFilterClearRssIn
+ * @sa otLinkFilterGetNextRssIn
  *
- * @sa otLinkAddWhitelist
- * @sa otLinkAddWhitelistRssi
- * @sa otLinkClearWhitelist
- * @sa otLinkGetWhitelistEntry
- * @sa otLinkIsWhitelistEnabled
- * @sa otLinkSetWhitelistEnabled
  */
-OTAPI void OTCALL otLinkRemoveWhitelist(otInstance *aInstance, const uint8_t *aExtAddr);
+OTAPI otError OTCALL otLinkFilterAddAddress(otInstance *aInstance, const otExtAddress *aExtAddress);
 
 /**
- * This function gets a MAC whitelist entry.
+ * This method adds an Extended Address to the filter and sets the RssIn for all the received messages from the
+ * Extended Address.
  *
- * @param[in]   aInstance A pointer to an OpenThread instance.
- * @param[in]   aIndex    An index into the MAC whitelist table.
- * @param[out]  aEntry    A pointer to where the information is placed.
+ * @param[in]  aInstance    A pointer to an OpenThread instance.
+ * @param[in]  aExtAddress  A reference to the Extended Address.
+ * @param[in]  aRss         The received signal strength to set.
+ *
+ * @retval OT_ERROR_NONE           Successfully added @p aExtAddress to the Filter and set @p aRss for @p aExtAddress.
+ * @retval OT_ERROR_INVALID_ARGS   If @p aExtAddress is NULL.
+ * @retval OT_ERROR_NO_BUFS        No available entry exists.
  *
- * @retval OT_ERROR_NONE          Successfully retrieved the MAC whitelist entry.
- * @retval OT_ERROR_INVALID_ARGS  @p aIndex is out of bounds or @p aEntry is NULL.
+ * @sa otLinkFilterGetAddressMode
+ * @sa otLinkFilterSetAddressMode
+ * @sa otLinkFilterAddAddress
+ * @sa otLinkFilterRemoveAddress
+ * @sa otLinkFilterClearAddresses
+ * @sa otLinkFilterGetNextAddress
+ * @sa otLinkFilterAddRssIn
+ * @sa otLinkFilterRemoveRssIn
+ * @sa otLinkFilterClearRssIn
+ * @sa otLinkFilterGetNextRssIn
  *
  */
-OTAPI otError OTCALL otLinkGetWhitelistEntry(otInstance *aInstance, uint8_t aIndex, otMacWhitelistEntry *aEntry);
+OTAPI otError OTCALL otLinkFilterAddAddressRssIn(otInstance *aInstance, const otExtAddress *aExtAddress, int8_t aRss);
```

Comment:
@jwhui , sorry for not explaining this in previous comments.

I am not planning to support rss relative spinel properties in this PR. So for windows drivers, I still want to use `IOCTL_OTLWF_OT_ADD_MAC_WHITELIST` to achieve this.

Another minor consideration is  two calls would cause two `FindEntry()` calls while only once if we keep this API.

If you prefer the split way, I could remove this API when supporting rss relative spinel properties in another separate PR?

Thoughts?